src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Mvc3/MvcRouteNamingHelper.cs

```diff
@@ -23,15 +21,15 @@ public static string TryGetControllerNameFromObject(ControllerContext controller
             return controllerType.Name;
         }
 
-        public static string TryGetActionNameFromRouteParameters(MethodCall methodCall, RouteData routeData)
+        public static string TryGetActionNameFromRouteParameters(MethodCall methodCall, dynamic routeData)
         {
             var actionName = methodCall.MethodArguments.ExtractAs<string>(1);
             if (actionName != null)
                 return actionName;
 
-            var directRouteMatches = routeData.Values.GetValueOrDefault("MS_DirectRouteMatches") as IEnumerable<RouteData> ?? Enumerable.Empty<RouteData>();
-            routeData = directRouteMatches.FirstOrDefault() ?? routeData;
-            actionName = routeData.Values.GetValueOrDefault("action") as string;
+            var directRouteMatches = routeData?.Values?["MS_DirectRouteMatches"] as IEnumerable<RouteData> ?? Enumerable.Empty<RouteData>();
+            routeData = directRouteMatches?.FirstOrDefault() ?? routeData;
+            actionName = routeData?.Values?["action"];
 
```

Comment:
Could have used original code here, but the 'default' of GetValueOrDefault would prevent actionName from being null and the method would return an empty string for the actionName, which doesn't look like the intended behavior.